lib/pure/cgi.nim

```diff
@@ -88,31 +80,8 @@ proc getEncodedData(allowedMethods: set[RequestMethod]): string =
 iterator decodeData*(data: string): tuple[key, value: TaintedString] =
```

Comment:
`template decodeData*(a: varargs[untyped]): untyped {.deprecated: "use uri.decodeData".}= uri.decodeData(a)`

or even better:
```
{.deprecated: [decodeData: uri.decodeData].}
```
(unless it hits https://github.com/nim-lang/Nim/issues/14819 in important_packages)

Reply:
There is no reason to deprecate `cgi.decodeData`

Reply:
having to ways to do the same thing is not good; you can keep a long deprecation cycle if needed.

Reply:
I agree on deprecating `cgi.decodeData`. But I think we should keep a long deprecation cycle. Because many nim repos use `cgi.decodeData`.

https://github.com/dom96/jester/search?q=decodedata
https://github.com/planety/prologue/search?q=decodedata
https://github.com/itsumura-h/nim-basolato/search?q=decodedata

`cgi.decodeData` is very common in net library.

Reply:
that's fine, we don't have to decide now how long deprecation cycle will be, what matters is to deprecate it at same time as the duplication is introduced

Reply:
Deprecating can be done after this PR is merged if someone is willing to deprecate it in the future.